### ISSUE TITLE
Custom context provider working when debugging Continue fails otherwise

### ISSUE THREAD

**Issue by frblondin** [continuedev/continue#789]
### Before submitting your bug report

- [X] I believe this is a bug. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug
- [X] I've seen the [troubleshooting guide](https://continue.dev/docs/troubleshooting) on the Continue Docs

### Relevant environment info

```Markdown
Continue: v0.9.35
Version: 1.85.2 (system setup)
Commit: 8b3775030ed1a69b13e4f4c628c612102e30a681
Date: 2024-01-18T06:40:10.514Z
Electron: 25.9.7
ElectronBuildId: 26354273
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.19045
```

### Description

I can successfully use and debug my custom context providers. When I no longer want to debug Continue and use the extension instead of the **Extension Run & Debug**, the custom providers returns the following error when executed:

![image](https://github.com/continuedev/continue/assets/5735134/07b1f95c-301e-463e-9100-6ce8a2dbd2b4)

... or for another provider:

![image](https://github.com/continuedev/continue/assets/5735134/40f06610-49a3-47ea-9f6c-5ee82b44538f)

From "Developer: Toggle Developer Tools":
![image](https://github.com/continuedev/continue/assets/5735134/386fdfc0-d658-4be9-aae0-88f5c0b2351e)

### To reproduce

1. Add following custom provider to `~/.continue` folder:
```typescript
const options = {
  maxResult: 50,
};

const commitSep = '<niouKommiT>';

const GitCommitContextProvider: CustomContextProvider = {
  title: "commits",
  displayTitle: "Commits",
  description: "Gets past commits.",
  type: "submenu",

  getContextItems: async (query: string, extras: ContextProviderExtras): Promise<ContextItem[]> => {
    const gitCommand = `git --no-pager log --max-count=${options.maxResult} --format="${commitSep}%s" --since="${query}"`
    const execSync = (await import('child_process')).default.execSync;
    const workspaceDirs = await extras.ide.getWorkspaceDirs();
    const output = execSync(gitCommand, { cwd: workspaceDirs[0] });
    return parseGitLog(output.toString());
  },

  async loadSubmenuItems(
    args: LoadSubmenuItemsArgs
  ): Promise<ContextSubmenuItem[]> {
    return [
      {
        title: "Commits from past week",
        description: "From past week",
        id: "1 week ago",
      },
      {
        title: "Commits from past month",
        description: "From past month",
        id: "1 month ago",
      },
      {
        title: "Commits from past 6 months",
        description: "From past 6 month",
        id: "6 months ago",
      },
      {
        title: "Commits from past year",
        description: "From past year",
        id: "1 year ago",
      },
    ];
  },
};

function parseGitLog(input:string): ContextItem[] {
  const result:ContextItem[] = [];
  input.split(commitSep).map(commit => {
    result.push( {
      name: commit,
      description: 'Commit message',
      content: '',
    })
  });
  return result;
}
```

2. Do `npm install child_process`
3. Try to use provider in Continue

### Log output

```Shell
My second problem is that I didn't manage to find any log file containing the message when doing Ctrl + Shift + P --> Open LogsFolder (or Open Extensions Logs Folder). [Troubleshooting](https://continue.dev/docs/troubleshooting) recommends to use "Developer: Toggle Developer Tools"... but it's limited to the browser, and providers are now supposed to be executed in NodeJS... but then why do I see my error in the console? In one word: confused. :-)
```

**Comment by sestinj** [comment]
@frblondin The console output in developer tools actually does show logs from NodeJS as well, so nothing wrong here.

As for the other errors I'm running myself and will update in a minute, but looks related to just some mismatch between environments in debug vs. extension modes

**Comment by sestinj** [comment]
Ok same things you said happened when I tried. Did a bit of digging around, and it looks like VS Code, or more likely Electron itself, is using `node:vm` and not setting [this parameter](https://nodejs.org/api/vm.html#:~:text=createCachedData().%0ADefault%3A%20false.-,importmoduledynamically%20%3Cfunction%3E%20called%20during%20evaluation%20of%20this%20module%20when%20import()%20is%20called.%20if%20this%20option%20is%20not%20specified%2C%20calls%20to%20import()%20will%20reject%20with%20err_vm_dynamic_import_callback_missing.,-This%20option%20is). This is something out of our control.

Luckily I'm working on a fix so you don't have to do dynamic imports (config.ts will run *only* in NodeJS, not the browser). I plan to spend the rest of today on shipping our first version of tab-autocomplete, but if you're willing to give me another day or two this will be done and there shouldn't be any problems.

And one other thing, the LLM will only see the `content` that you specify, so I would change to `content: commit` in the objects you are returning from `parseGitLog`.

<img width="1512" alt="Screen Shot 2024-01-30 at 10 50 30 AM" src="https://github.com/continuedev/continue/assets/33237525/b43ec8fe-a486-4f22-99dd-e70e69e90ce7">

**Comment by sestinj** [comment]
oh wait! Actually I found a solution: `const execSync = (await require("child_process")).execSync;`

Using require instead of import worked for me.

Everything else I said above is still true, I hope to let you just use normal imports within the next couple of days

**Comment by frblondin** [comment]
I confirm that provided workaround (with require) works. Thanks for working on using NodeJS in all cases!

I suggest to keep this issue open until you complete this work.

**Comment by sestinj** [comment]
@frblondin The work in moving to NodeJS is now done and available in pre-release 0.9.47.

I've confirmed that I could use `import { execSync } from "child_process";` at the top of the file, but I'll wait to close this issue until I know it's good for you as well

**Comment by frblondin** [comment]
I confirm. Brillant!
